proxy/connect/connect.go

```diff
@@ -45,8 +48,8 @@ type Destination interface {
 }
 
 type SendRequest struct {
-	Metric       *metricpb.Metric
-	ErrorChannel chan error
+	Metric    *metricpb.Metric
+	Timestamp time.Time
```

Comment:
While it will be very interesting to see how long metrics stay in the buffer, it may be worth separating this change from the rest of the PR just to limit complexity per unit of change.

Duration of request and buffer enqueue error count are IMO enough info for now